changelog.md

```diff
@@ -104,6 +104,9 @@
 - In `std/dom`, `Interval` is now a `ref object`, same as `Timeout`. Definitions of `setTimeout`,
   `clearTimeout`, `setInterval`, `clearInterval` were updated.
 
+- `os.parseCmdLine` with `-d:nimPreviewParseCmdLine` (enabled on devel) now raises on posix for invalid inputs,
```

Comment:
what are these "invalid inputs" and isn't it better when command line parsing cannot fail? Just pick some interpretation for quirky inputs and be done with it, I think hardly anybody knows how to handle these parsing failures...

Reply:
> what are these "invalid inputs"

I've now specified what invalid inputs are in the doc comment for `parseCmdLine` (unclosed single/double quotes, or unfinished escape sequence).

It's consistent with python which raises:
```py
import shlex; list(shlex.shlex('ab"', posix=True)) # ValueError: No closing quotation
import shlex; list(shlex.shlex('ab\\', posix=True)) # ValueError: No escaped character
import shlex; list(shlex.shlex("ab'", posix=True)) # ValueError: No closing quotation
```

the shell itself also raises on unclosed single or double quote:
```nim
when defined case9:
  {.emit:"""
#include <stdio.h>
#include <stdlib.h>
void fn() {
  int ret = system("echo abc'");
  printf("ret: %d\n", ret);
  int ret2 = system("echo abc\"");
  printf("ret2: %d\n", ret2);
}
  """.}
  proc fn(){.importc.}
  fn()
```
```
sh: -c: line 0: unexpected EOF while looking for matching `''
sh: -c: line 1: syntax error: unexpected end of file
ret: 512
sh: -c: line 0: unexpected EOF while looking for matching `"'
sh: -c: line 1: syntax error: unexpected end of file
ret2: 512
```

> isn't it better when command line parsing cannot fail? Just pick some interpretation for quirky inputs and be done with it, I think hardly anybody knows how to handle these parsing failures...

masking errors only leads to more buggy software and harder to track bugs where the effect occurs somewhere down the line instead of at the place where the bug should've been detected. Trying to assign a value on invalid input (eg `"abc\"d ef".parseCmdLine`) is arbitrary, and passing this to a shell would fail anyways.